src/components/PostDetails.tsx

```diff
@@ -1,117 +1,212 @@
-import React from 'react';
+import {
+  FC,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { getComments, deleteComment, addComment } from '../api/comments';
+import { Post } from '../types/Post';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment, CommentData } from '../types/Comment';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [isError, setError] = useState(false);
+  const [isFormOpen, setFormOpen] = useState(false);
+  const [isAddingError, setAddingError] = useState(false);
+  const [isAddingLoading, setAddingLoadig] = useState(false);
+  const [commentField, setCommentField] = useState('');
+  const [commentFieldError, setCommentFieldError] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    setError(false);
+
+    const fetchData = async () => {
+      try {
+        const commentsFromServer = await getComments(post.id);
+
+        setComments(commentsFromServer);
+      } catch {
+        setError(true);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, [post]);
+
+  useEffect(() => {
+    const handlePostButtonClick = () => {
+      setFormOpen(false);
+    };
+
+    const buttons = document.querySelectorAll('[data-cy="PostButton"]');
+
+    buttons.forEach(button => {
+      button.addEventListener('click', handlePostButtonClick);
+    });
```

Comment:
`data-cy="PostButton"` is only for cypress, don't use that in your app code.

Just [lift your state up](https://react.dev/learn/sharing-state-between-components#lifting-state-up-by-example).

But good job for being responsible and calling `button.removeEventListener` :fire: